core/src/test/scala/org/apache/spark/util/UtilsSuite.scala

```diff
@@ -35,7 +36,64 @@ import org.apache.hadoop.fs.Path
 import org.apache.spark.SparkConf
 
 class UtilsSuite extends FunSuite with ResetSystemProperties {
+  
+  test("timeConversion") {
+    // Test -1
+    assert(Utils.timeStringAsSec("-1") === -1)
```

Comment:
do we actually want this behavior? I would think that if the user tries to give a negative value to a time config we should fail loudly.